File: python/228_Summary_Ranges/solution.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return []
        start, prev = nums[0], nums[0]
        answer = []
        for i in range(1, len(nums)):
            if nums[i] != prev + 1:
                if start == prev:
                    answer.append(f'{start}')
                else:
                    answer.append(f'{start}->{prev}')
                start = nums[i]
            prev = nums[i]
        if start == prev:
            answer.append(f'{start}')
        else:
            answer.append(f'{start}->{prev}')
        return answer

File: rust/228_Summary_Ranges/solution.rs
impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut answer: Vec<String> = Vec::new();
        if nums.is_empty() {
            return answer;
        }
        let mut prev = nums[0];
        let mut start = nums[0];
        for i in 1..nums.len() {
            if nums[i] != prev + 1 {
                if prev != start {
                    answer.push(format!("{}->{}", start, prev));
                } else {
                    answer.push(format!("{}", start));
                }
                start = nums[i];
            }
            prev = nums[i]
        }
        if prev != start {
            answer.push(format!("{}->{}", start, prev));
        } else {
            answer.push(format!("{}", start));
        }
        answer
    }
}
